tsi-typescript.el

```diff
@@ -370,42 +370,61 @@
      (or comment-indentation 0))))
 
 (defun tsi--current-line-empty-p ()
+  "Internal function.
+
+Returns true if the line at point is empty or nothing but whitespace."
   (string-match-p "\\`[[:space:]]*$" (thing-at-point 'line)))
 
+(defun tsi--node-line-num (node)
+  "Internal function.
+
+Return the line number of the start point of the given node."
+  (car (tsc-node-start-point node)))
+
+(defconst tsi-typescript--doubly-nestable-types '(array object statement_block)
+  "List of node  types that can be doubly nested on the same like.
+Used to avoid double indenting blank lines following things like '[{'")
```

Comment:
this should do the right thing for multiple nested data structures (e.g. `[{[`), right?

Reply:
Good question. I just verified manually, and it does.

I will add a test or two to cover that scenario - might be a day or two.